ftl/python/builder.py

```diff
@@ -27,82 +28,132 @@
 
 _PYTHON_NAMESPACE = 'python-requirements-cache'
 _REQUIREMENTS_TXT = 'requirements.txt'
+_WORK_DIRS = ['app', 'env', 'wheel']
 
 
 class Python(builder.JustApp):
     def __init__(self, ctx):
         self.descriptor_files = [_REQUIREMENTS_TXT]
         self.namespace = _PYTHON_NAMESPACE
+        self.ctx = ctx
         super(Python, self).__init__(ctx)
 
     def __enter__(self):
         """Override."""
         return self
 
-    def _generate_overrides(self):
+    def _generate_overrides(self, set_path):
+        env = {
+            "VIRTUAL_ENV": "/env",
+        }
+        if set_path:
+            env['PATH'] = '/env/bin:$PATH'
         return metadata.Overrides(
-            creation_time=str(datetime.date.today()) + "T00:00:00Z")
+            creation_time=str(datetime.date.today()) + "T00:00:00Z", env=env)
 
-    def CreatePackageBase(self, base_image, cache):
+    def CreatePackageBase(self, base_image, py_version='python2.7'):
         """Override."""
-        overrides = self._generate_overrides()
-
-        layer, sha = self._gen_package_tar()
-        logging.info('Generated layer with sha: %s', sha)
+        package_base = base_image
+
+        dir_map = self._gen_dirs(_WORK_DIRS)
```

Comment:
Actually the map seems a little awkward here - what about just setting these on `self` and keeping the list of all dirs to create on self as well?

Reply:
done